addons/html_editor/static/src/core/clipboard_plugin.js

```diff
@@ -600,6 +601,48 @@ export class ClipboardPlugin extends Plugin {
         if (!isHtmlContentSupported(ev.target)) {
             return;
         }
+        let caretRange;
+        if (this.document.caretPositionFromPoint) {
+            const range = this.document.caretPositionFromPoint(ev.clientX, ev.clientY);
+            caretRange = document.createRange();
+            caretRange.setStart(range.offsetNode, range.offset);
+            caretRange.setEnd(range.offsetNode, range.offset);
+        } else if (this.document.caretRangeFromPoint) {
+            caretRange = this.document.caretRangeFromPoint(ev.clientX, ev.clientY);
+        }
+        if (caretRange) {
+            const caretRect = caretRange.getBoundingClientRect();
+
+            const selection = this.dependencies.selection.getEditableSelection();
+            const range = document.createRange();
+            range.setStart(selection.anchorNode, selection.anchorOffset);
+            range.setEnd(selection.focusNode, selection.focusOffset);
+            const selectionRect = range.getBoundingClientRect();
+
+            // Check if the caret's bounding box is inside the selection's bounding box
+            if (
+                caretRect.left >= selectionRect.left &&
+                caretRect.right <= selectionRect.right &&
+                caretRect.top >= selectionRect.top &&
+                caretRect.bottom <= selectionRect.bottom
+            ) {
+                return true; // Do nothing if the drop is inside the selection
+            }
+            const nodeToSplit =
+                selection.direction === DIRECTIONS.RIGHT
+                    ? selection.focusNode
+                    : selection.anchorNode;
+            const offsetToSplit =
+                selection.direction === DIRECTIONS.RIGHT
+                    ? selection.focusOffset
+                    : selection.anchorOffset;
+            if (nodeToSplit.nodeType === Node.TEXT_NODE && !selection.isCollapsed) {
+                const selectionToRestore = this.dependencies.selection.preserveSelection();
+                splitTextNode(nodeToSplit, offsetToSplit, DIRECTIONS.LEFT);
+                selectionToRestore.restore();
+            }
```

Comment:
I don't understand why this is needed either.

Reply:
I am splitting the text before determining the drop position to ensure that the offset find using caret does not exceed the node size after the selection is deleted.